lamindb/core/storage/paths.py

```diff
@@ -139,7 +139,20 @@ def store_file_or_folder(
     local_path = UPath(local_path)
     if not isinstance(storage_path, LocalPathClasses):
         # this uploads files and directories
-        create_folder = False if local_path.is_dir() else None
+        if local_path.is_dir():
+            create_folder = False
+            try:
+                # if storage_path already exists we need to delete it
+                # if local_path is a directory
+                # to replace storage_path correctly
+                if storage_path.stat().as_info()["type"] == "directory":
```

Comment:
It seems odd that neither line 149 nor 151 are covered by a test. Isn't this here core functionality introduced in the PR. I would have assumed that the add-replace-cache tests capture this.

Reply:
Yes, indeed, very strange, I will check locally what is going on.

Reply:
Ok, i was using a different suffix, so the deletion wasn't tested indeed, as the replacement had a different path. Fix is here https://github.com/laminlabs/lamindb/pull/2381